app/models/pawn.rb

```diff
@@ -20,6 +20,49 @@ def promote!(new_type)
 
   private
 
+  # En Passant Logic:
+  # 1)target pawn just moved 2 space forward
+  # 2)only capture within the next turn
+  # 3)can only capture by opponent's pawn
+  # 4)opponent pawn must be on the left or right of target piece's y axis
+
+  # En Passant Capture logic
+  def en_passant_capture?(x, y)
+    return true if en_passant_possible?(x, y) || en_passant_y_diff?(x, y)
+  end
+
+  # Return false if victim piece is nil or same color
+  # Return true  if victim piece's y axis difference is equal to 1
+  def en_passant_y_diff?(x, y)
+    victim = last_moved_piece
+    return false if victim.nil? || victim.color == color
+    return true if  x == victim.x_position &&
+                    (y == victim.y_position - 1 || y == victim.y_position + 1)
+  end
+
+  # Return false if last move piece is nil (Start of game)
+  # Return false last moved piece is not pawn or its y axis difference is !2
+  # Check both both adjacent side of pawn
+  # If any have valid En Passant pawn piece, return true
+  def en_passant_possible?(x, y)
+    return false if last_moved_piece.nil?
+    return false unless last_moved_piece.y_distance(y - 2) == 2 &&
+                        type == 'Pawn'
```

Comment:
If I understand this correctly, you moved y_distance to a public method and are now invoking it on the last_moved_piece, which would be the pawn victim.  But if you pass in the coordinate of where pawn is going to "attack" as the coordinate behind where the victim is now, I don't see how this logic could work.  If victim is at 3,3 and aggressor pawn is at 4,3, calling this method on 3,2 which is where aggressor pawn would 'attack' would fail this check.   It would resolve as:

abs(victim.y_position - y)
abs(3 - 0)
abs(3)
3